src/gwt/src/org/rstudio/studio/client/workbench/views/source/SourceColumnManager.java

```diff
@@ -1934,8 +1982,24 @@ public void initVimCommands()
       vimCommands_.addStarRegister();
    }
 
+   public SourceAppCommand getSourceCommand(AppCommand command, SourceColumn column)
+   {
+      // check if we've already create a SourceAppCommand for this command
+      for (SourceAppCommand cmd : sourceAppCommands_)
+      {
+         if (cmd.getCommand() == command &&
```

Comment:
Testing JavaScript objects for equality has a bunch of weird edge cases so I'd recommend avoiding it. For example the object could be cloned and the clone won't match, but they'll look like the same object in the debugger so it won't be obvious WHY they don't match!

How about storing `sourceAppCommands_` as a `HashMap`, using the command ID (from `command.getId()`) plus the column name as a key? Then lookups will be faster, plus identity will be more reliable as you'll just need to check for existence in the map.

Reply:
Thank you! I've implemented the update 👍